locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/templates/customize/TemplateOptionsOption.java

```diff
@@ -39,8 +49,22 @@ public void apply(TemplateOptions options, ConfigBag config, Object v) {
 
         Class<? extends TemplateOptions> clazz = options.getClass();
         for (final Map.Entry<String, Object> option : optionsMap.entrySet()) {
-            if (option.getValue() != null) {
-                Maybe<?> result = MethodCoercions.tryFindAndInvokeBestMatchingMethod(options, option.getKey(), option.getValue());
+            Object optionValue = option.getValue();
+            if (optionValue != null) {
+
+                try {
+                    final EntityInternal entity = (EntityInternal) BrooklynTaskTags.getTargetOrContextEntity(Tasks.current());
+                    final ExecutionContext exec = (null != entity
+                        ? entity.getExecutionContext()
+                        : BasicExecutionContext.getCurrentExecutionContext());
+                    if (exec != null) {
```

Comment:
I wonder if there is some way we could log.warn if `exec != null` and if the value is a deferred supplier. But not sure how easy it is to do for the latter (looking at `ValueResolver`, it handles `v instanceof ImmediateSupplier || v instanceof DeferredSupplier || v instanceof TaskFactory || v instanceof TaskAdaptable || v instanceof Task || v instanceof Future`, but it also recursively looks inside iterables and maps.

Therefore maybe just ignore this comment :-(